pennylane/qnodes/base.py

```diff
@@ -600,5 +604,10 @@ def evaluate_obs(self, obs, args, kwargs):
         self._set_variables(args, kwargs)
 
         self.device.reset()
-        ret = self.device.execute(self.circuit.operations, obs, self.circuit.variable_deps)
+
+        if isinstance(self.device, qml.QubitDevice):
+            circuit_graph = CircuitGraph(self.circuit.operations + obs, self.circuit.variable_deps)
```

Comment:
A comment might be useful here why ``self.circuit.operations + obs`` is passed as the first argument